packages/integration-tests/Executions.js

```diff
@@ -30,11 +32,31 @@ const findExecutionArn = async (prefix, matcher, options = { timeout: 0 }) =>
   pRetry(
     async () => {
       let execution;
+      let pageNumber = 1;
 
       try {
-        const { body } = await executionsApi.getExecutions({ prefix });
-        const executions = JSON.parse(body);
+        const { body } = await executionsApi.getExecutions({
+          prefix,
+          query: {
+            limit: EXECUTION_LIST_LIMIT,
+            page: pageNumber
+          }
+        });
+        let executions = JSON.parse(body);
         execution = executions.results.find(matcher);
+
+        while (isNil(execution) && executions.meta.count > (EXECUTION_LIST_LIMIT * pageNumber)) {
+          // eslint-disable-next-line no-await-in-loop
+          const response = await executionsApi.getExecutions({
+            prefix,
+            query: {
+              limit: EXECUTION_LIST_LIMIT,
+              page: pageNumber += 1
+            }
+          });
+          executions = JSON.parse(response.body);
+          execution = executions.results.find(matcher);
```

Comment:
Similarly, I wonder here if we couldn't query ES directly with the properties we're expecting in the `matcher`. I suppose if this is working it's not essential though

Reply:
Yeah so this one gets a bit more complicated. I looked into it and there are some instances where we're trying to match on `originalPayload.testExecutionId` which I think would be impossible through the field queries without a more advanced ES query.